electrum/lnpeer.py

```diff
@@ -1050,27 +981,108 @@ def are_datalossprotect_fields_valid() -> bool:
                 pass
             else:
                 if our_remote_pcp != their_local_pcp:
-                    self.logger.error(f"channel_reestablish ({chan.get_id_for_log()}): "
-                                      f"(DLP) remote PCP mismatch: {bh2u(our_remote_pcp)} != {bh2u(their_local_pcp)}")
+                    self.logger.error(
+                        f"channel_reestablish ({chan.get_id_for_log()}): "
+                        f"(DLP) remote PCP mismatch: {bh2u(our_remote_pcp)} != {bh2u(their_local_pcp)}")
                     return False
             return True
-
         if not are_datalossprotect_fields_valid():
             raise RemoteMisbehaving("channel_reestablish: data loss protect fields invalid")
-
         if they_are_ahead:
-            self.logger.warning(f"channel_reestablish ({chan.get_id_for_log()}): "
-                                f"remote is ahead of us! They should force-close. Remote PCP: {bh2u(their_local_pcp)}")
+            self.logger.warning(
+                f"channel_reestablish ({chan.get_id_for_log()}): "
+                f"remote is ahead of us! They should force-close. Remote PCP: {bh2u(their_local_pcp)}")
             # data_loss_protect_remote_pcp is used in lnsweep
             chan.set_data_loss_protect_remote_pcp(their_next_local_ctn - 1, their_local_pcp)
             self.lnworker.save_channel(chan)
             chan.peer_state = PeerState.BAD
-            return
-        elif we_are_ahead:
+            raise RemoteMisbehaving("remote ahead of us")
```

Comment:
What if we haven't sent `channel_reestablish` yet? If we immediately disconnect, the remote won't know they are ahead.

Reply:
right. note that I replaced the `Future` with a `Queue`
To address that case, we could revert to a `Future` and use `set_exception` instead of raising

Reply:
ok, I guess that's a convincing argument for a Future here.
It also justifies breaking the homogeneity between the error cases; we could `set_exception` on this line and 4 lines below, and add a comment that it is to wait until `channel_reestablish` is sent.

Reply:
fixed in 395911b08651639593c9f90a3401f0c19b6d23a7